### PR TITLE
Make initWithValue: create standalone copies of links & arrays

### LINKED ISSUE #1924: Test making standalone copies of an object graph via the standalone initializer
Was done in https://github.com/realm/realm-cocoa/issues/2043, which was never merged

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1924.

\c @tgoyne @jpsim @bdash

### CODE REVIEW

**Comment by jpsim** [comment]
retest this please

**Comment by segiddins** [comment]
review this please?

**Comment by jpsim** [comment]
This looks good, but it'd be nice to have a Swift performance test for this too. Don't forget to update the Xcode baselines and add a CHANGELOG entry.

**Comment by segiddins** [comment]
Changelog added, this should be good to go

**Comment by segiddins** [comment]
ping for review?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ x.xx.x Release notes (yyyy-MM-dd)
 ### Enhancements
 
 * Reduce the amount of memory used by RLMRealm notification listener threads.
+* Entire standlone object graph copies can now be made with
```
Comment (jpsim) [inline CHANGELOG.md]: standalone

**Comment by jpsim** [comment]
retest this please

**Comment by segiddins** [comment]
This is rebased.

**Comment by tgoyne** [comment]
We need to discuss if this is something we want to do with the Java guys.

**Comment by jpsim** [comment]
This PR adds what I think is the appropriate (and expected) behavior. @segiddins can you please rebase this so we can revive this PR?

**Comment by segiddins** [comment]
Rebased, @jpsim.

**Comment by jpsim** [comment]
Looks like the baseline needs to be updated since rebasing:

```
PerformanceTests.testStandaloneObjectInitialization, failed: Time average is 21% worse (max allowed: 10%).
```

**Comment by segiddins** [comment]
Fixed

**Comment by jpsim** [comment]
This looks good to me, I understand others may have reservations about the fundamental necessity of this altogether, but there are certain legitimate scenarios that this change greatly simplifies.

:+1:

**Comment by tgoyne** [comment]
I'm still not convinced there are use-cases for this that aren't just working around other missing features or anti-patterns, and this isn't a place where we can do something better than what users will write on their own with no downsides.

**Comment by segiddins** [comment]
At this point, should I just close this PR? I'd still like to see it merged, but I understand others might have stronger feelings on this than I.

**Pull request closed without merging**